lib/registration.pm

```diff
@@ -547,6 +547,24 @@ sub process_scc_register {
     elsif (!get_var('SCC_REGISTER', '') =~ /addon|network/) {
         send_key $cmd{next};
     }
+
+    # Some licenses (like HA) can be shown on sle 15 after registration
+    if (check_screen('scc-downloading-license', timeout => 5) && is_sle('15+')) {
+        my @tags = qw(license-agreement yast_scc-automatic-changes yast_scc-installation-summary yast_scc-pkgtoinstall);
+        while (1) {
+            assert_screen(\@tags, timeout => 240);
+            send_key $cmd{ok}     if match_has_tag('yast_scc-automatic-changes');
+            send_key $cmd{accept} if match_has_tag('yast_scc-pkgtoinstall');
+            if (match_has_tag('yast_scc-installation-summary')) {
+                send_key $cmd{finish};
+                last;
+            }
+            if (match_has_tag('license-agreement')) {
+                y2_logs_helper::accept_license;
+                send_key $cmd{next};
+            }
+        }
+    }
```

Comment:
sorry if I wasn't clear enough. By "inside process_scc_register" I did not mean to simply move it there but follow the implementation that has been used so far, i.e. move it inside "process_scc_register_addons" or within the if above to not apply for registration after installation, right?. Inside there within "if (get_var('SCC_ADDONS'))", within there probably you want to make it depend on the specific addons. And because the method because very long and hard to maintain this is a good opportunity to refactor. There is no better time than whenever someone adds new changes, so the job falls onto you, I am afraid.

Also, as you probably know by now, please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

We do not want to waste a waiting time of 5s for every single job that does not have this special case, right?

Reply:
In our cases we are not using `SCC_ADDONS` nor `SCC_REGISTER` here, because it's an upgrade from DVD media, so `process_scc_register_addons` is never used here and we don't know in advance which addons are already installed. But the registration should occur during `patch_sle` because the "old" OS should be up-to-date before upgrade. So we "simply" re-register the server with no new addons, but HA now show the license during re-registration (maybe other modules also, but I only have it with HA for now).

But I will try to refactor the code and find a way to handle this in the same code.